data/TaskDAO.js

```diff
@@ -0,0 +1,70 @@
+const Task = require('./Task');
+// In-memory implementation
+class TaskDAO {
+    constructor() {
+        this.tasks = [];
+        this.maxID = 0;
+    }
+
+    find(criteria) {
+        if(!criteria || Object.keys(criteria).length === 0) return this.tasks;
+        else {
+            return this.tasks.filter(task => this.matchesCriteria(task, criteria));
+        }
+    }
+
+    findById(id) {
+        return this.tasks.filter(task => task.id == id);
+    }
+
+    newTask(props) {
+        if(!this.validateNewTaskProps(props)) return;
+        let taskID = ++this.maxID;
```

Comment:
these should both be `const` and not `let`. With a very small scope it would be okay to put the new task inside of the `push`. You could even `++this.maxID` inline. It's fine to leave it out like this though. The scope of the method has a big impact on what's acceptable.